File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int count = 0;
    
    do
    {
        printf("Price in dollars: ");
        dollars = get_float();
    }
    while(dollars < 0);
    
    int cents = round(dollars * 100);
    
    while(cents >= 25) { count += 1; cents = cents - 25; }
    
    while(cents >= 10) { count += 1; cents = cents - 10; }
    
    while(cents >= 5) { count += 1; cents = cents - 5; }
    
    while(cents >= 1) { count += 1; cents = cents - 1; }
    
    printf("%i\n", count);

}

File: pset6/greedy.py
import cs50

class counter:
    def __init__(self, cents, count):
        self.cents = cents
        self.count = count
    
    def increment():
        counter.count += 1
        

def main():
    counter.count = 0
    while True:
        print("Price in dollars: ", end="")
        dollars = cs50.get_float()
        if dollars > 0:
            break
    
    counter.cents = round(dollars * 100)
    
    money(25)
    money(10)
    money(5)
    money(1)
    
    print(counter.count)
    
def money(n):
    while counter.cents >= n:
        counter.increment()
        counter.cents = counter.cents - n
    
if __name__ == "__main__":
    main()
